### ISSUE TITLE
[Bug?]: `@yarnpkg/plugin-version < 3.1.0-rc.2` is not compatible with `@yarnpkg/cli >= 3.1.0-rc.12`

### ISSUE THREAD

**Issue by cuyl** [yarnpkg/berry#3615]
### Self-service

- [ ] I'd be willing to implement a fix

### Describe the bug

`@yarnpkg/plugin-version@3.1.0-rc.2` is not compatible with `@yarnpkg/cli < 3.1.0-rc.12`
`@yarnpkg/plugin-version < 3.1.0-rc.2` is not compatible with `@yarnpkg/cli >= 3.1.0-rc.12`

### To reproduce

```js repro
const util = require('util');
const _exec = util.promisify(require('child_process').exec);
const _execFile = util.promisify(require('child_process').execFile);

const env = {...process.env, YARN_ENABLE_COLORS: 0, YARN_IGNORE_PATH: 1, YARN_ENABLE_INLINE_BUILDS: 1, YARN_ENABLE_IMMUTABLE_INSTALLS: 0};

const exec = async cmd => {
  let stdout
  try {
    ({stdout} = await _exec(cmd, { env }));
  } catch (error) {
    error.message = error.stdout
    throw error
  }
  return stdout
};

const execFile = async (...args) => {
  let stdout
  try {
    ({stdout} = await _execFile(process.execPath,[...args], { env }));
  } catch (error) {
    error.message = error.stdout
    throw error
  }
  return stdout
};

await packageJson({});
await exec('yarn set version 3.1.0-rc.11');

await expect(execFile(...'./.yarn/releases/yarn-3.1.0-rc.11.cjs plugin import @yarnpkg/plugin-version'.split(' '))).resolves.toBeTruthy();
await expect(execFile(...'./.yarn/releases/yarn-3.1.0-rc.11.cjs set version 3.1.0-rc.12'.split(' '))).resolves.toBeTruthy();
await expect(execFile(...'./.yarn/releases/yarn-3.1.0-rc.12.cjs plugin import @yarnpkg/plugin-version'.split(' '))).resolves.toBeTruthy();
```

### Environment

```shell
System:
    OS: macOS 10.15.7
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
  Binaries:
    Node: 12.22.7 - /private/var/folders/x9/m_y2fpmn1wgcd33f1y0tdhgr0000gq/T/xfs-3eedbbd3/node
    Yarn: 3.1.0-rc.12 - /private/var/folders/x9/m_y2fpmn1wgcd33f1y0tdhgr0000gq/T/xfs-3eedbbd3/yarn
    npm: 6.14.15 - ~/.nvm/versions/node/v12.22.7/bin/npm
  npmPackages:
    jest: ^27.3.1 => 27.2.0
```

### Additional context
Errlog
```
Error: Internal Error: Cannot redefine settings "changesetBaseRefs"
```

**Comment by rachel-church** [comment]
I also encountered this bug and any attempt to upgrade yarn to v3.1.0 and plugin-version to v3.1.0 resulted in the following error for me:
```
Internal Error: Cannot redefine settings "changesetBaseRefs"
    at Ra.importSettings (/Users/rachelchurch/src/marketing/.yarn/releases/yarn-3.1.0.cjs:432:26048)
    at Ra.activatePlugin (/Users/rachelchurch/src/marketing/.yarn/releases/yarn-3.1.0.cjs:432:25923)
    at Function.find (/Users/rachelchurch/src/marketing/.yarn/releases/yarn-3.1.0.cjs:432:24189)
    at async i (/Users/rachelchurch/src/marketing/.yarn/releases/yarn-3.1.0.cjs:448:11019)
    at async r (/Users/rachelchurch/src/marketing/.yarn/releases/yarn-3.1.0.cjs:448:10617)
```

### Workaround

1. Manually edit the `.yarnrc.yml` file manually and remove the `@yarnpkg/plugin-version` from the `plugins` list and any configuration options specific to `plugin-version`
2. Update yarn with `yarn set version 3.1.x` or a similar `yarn set version` command
3. Upgrade the versions plugin with `yarn plugin import @yarnpkg/plugin-version@3.1.0-rc.3`
4. Manually edit the `.yarnrc.yml` file to add back your versions plugin configuration options

**Comment by lensbart** [comment]
Seems to be fixed in latest (3.1.1)
